libcxx/include/string

```diff
@@ -3349,70 +3371,48 @@ inline _LIBCPP_CONSTEXPR_SINCE_CXX20 void basic_string<_CharT, _Traits, _Allocat
   if (__target_capacity == capacity())
     return;
 
```

Comment:
We should add `_LIBCPP_ASSERT_INTERNAL(__is_long())` here, and then the `if (!__is_long())` below can go away.